wormhole_chain/x/wormhole/keeper/vaa.go

```diff
@@ -47,3 +50,59 @@ func (k Keeper) VerifyVAA(ctx sdk.Context, vaa *vaa.VAA) error {
 
 	return nil
 }
+
+// Verify a governance VAA:
+// - Check signatures
+// - Replay protection
+// - Check the source chain and address is governance
+// - Check the governance payload is for wormchain and the specified module
+// - return the parsed action and governance payload
+func (k Keeper) VerifyVAAGovernance(ctx sdk.Context, v *vaa.VAA, module [32]byte) (action byte, payload []byte, err error) {
+	if err = k.VerifyVAA(ctx, v); err != nil {
+		return
+	}
+	_, known := k.GetReplayProtection(ctx, v.HexDigest())
+	if known {
+		err = types.ErrVAAAlreadyExecuted
+		return
+	}
+	// Prevent replay
+	k.SetReplayProtection(ctx, types.ReplayProtection{Index: v.HexDigest()})
```

Comment:
I think this solution is better because it's a bit easier to read and harder for the caller to mess up.  It's functionally the same, in that if the subsequent action isn't successful, an error should be returned and changes will be rolled back.